docs/docs/markers.mdx

```diff
@@ -0,0 +1,289 @@
+Markers allow you to describe conditions over such events, and when these conditions are met, the dialogues are "marked" for further analysis or inspection.
+
+There are several applications for Markers. For example, they can be used to define your bot's **Key Performance Indicators (KPIs)**,
+such as dialogue completion or task success. Take [Carbon Bot](https://rasa.com/blog/using-conversation-tags-to-measure-carbon-bots-success-rate/)
+for example, which helps users offset their carbon emissions from flying. For Carbon Bot, you might define **dialogue completion** as "all mandatory slots have been filled",
+and **task success** as "all mandatory slots have been filled and a carbon estimate has been successfully computed".
+Marking when these important events occur allows you to quantify Carbon Bot's success rate.
+
+Markers also allow you to **diagnose your dialogues** by surfacing important events for further inspection.
+For example, you might observe that Carbon Bot tends to successfully set the `travel_departure` and `travel_destination` slots,
+but fails to set the `travel_flight_class` slot. You can define a marker to quantify how often this behavior occurs
+and surface relevant dialogues for review as part of
+[Conversation Driven Development (CDD)](./conversation-driven-development.mdx).
+
+Marker definitions are written in `YAML`. For example, here are the markers that define dialogue completion and task success for Carbon Bot:
+
+```yaml
+marker_dialogue_completion:
+  and:
+    - slot_was_set: travel_departure
+    - slot_was_set: travel_destination
+    - slot_was_set: travel_flight_class
+
+marker_task_success:
+  and:
+    - slot_was_set: travel_departure
+    - slot_was_set: travel_destination
+    - slot_was_set: travel_flight_class
+    - action: provide_carbon_estimate
+```
+
+And here is the marker for surfacing dialogues where all mandatory slots are set except `travel_flight_class`:
+
+```yaml
+marker_dialogue_mandatory_slot_failure:
+  and:
+    - slot_was_set: travel_departure
+    - slot_was_set: travel_destination
+    - not:
+      - slot_was_set: travel_flight_class
+```
+
+The next sections explain how write marker definitions, how to apply them to your existing dialogues, and what the output format looks like.
+
+## Defining Markers
+
+You define you markers in a marker configuration file written in `YAML`.
+Each marker should have a **unique identifier**, and consists of at least one **event condition**.
+Markers can also contain **operators**, which allow you to express more nuanced behavior or
+combine event conditions.
+
+Consider the following marker definition:
+```yaml
+marker_mood_expressed:
+  or:
+    - intent: mood_unhappy
+    - intent: mood_great
+```
+The unique marker identifier is `marker_mood_expressed`. This marker definition contains one operator `or`,
+and two event conditions `intent: mood_unhappy` and `intent: mood_great`.
+
```

Comment:
Maybe add a sentence explaining what this means, i.e. that the `steedmarker` is triggered when _either_ `mood unhappy` is detected _or_ `mood great` is detected.